oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServletOAuth2AuthorizedClientExchangeFilterFunction.java

```diff
@@ -268,11 +289,41 @@ public void setAccessTokenExpiresSkew(Duration accessTokenExpiresSkew) {
 	public Mono<ClientResponse> filter(ClientRequest request, ExchangeFunction next) {
 		Optional<OAuth2AuthorizedClient> attribute = request.attribute(OAUTH2_AUTHORIZED_CLIENT_ATTR_NAME)
 				.map(OAuth2AuthorizedClient.class::cast);
-		return Mono.justOrEmpty(attribute)
-				.flatMap(authorizedClient -> authorizedClient(request, next, authorizedClient))
-				.map(authorizedClient -> bearer(request, authorizedClient))
-				.flatMap(next::exchange)
-				.switchIfEmpty(next.exchange(request));
+		if (!attribute.isPresent()) {
```

Comment:
This code is the same as the else block. It should be able to be restructured like: try and get the authorized client, if not there default it, process (same process in either case).

Reply:
Applied in latest commit